src/middleware.ts

```diff
@@ -1,24 +1,27 @@
-import { EventEmitter } from 'events'
+import { StrictEventEmitter } from 'strict-event-emitter'
 import { Headers } from 'headers-polyfill'
 import { RequestHandler as ExpressMiddleware } from 'express'
-import { RequestHandler, handleRequest, parseIsomorphicRequest } from 'msw'
+import { RequestHandler, handleRequest, MockedRequest } from 'msw'
+import { encodeBuffer } from '@mswjs/interceptors'
 
-const emitter = new EventEmitter()
+const emitter = new StrictEventEmitter()
```

Comment:
https://github.com/mswjs/http-middleware/runs/7319161973?check_suite_focus=true

```

src/middleware.ts:32:7 - error TS2345: Argument of type 'EventEmitter' is not assignable to parameter of type 'StrictEventEmitter<ServerLifecycleEventsMap>'.
      The types returned by 'eventNames()' are incompatible between these types.
        Type '(string | symbol)[]' is not assignable to type 'Event[]'.
          Type 'string | symbol' is not assignable to type 'Event'.
            'Event' could be instantiated with an arbitrary type which could be unrelated to 'string | symbol'.
```

It seems that `StrictEventEmitter` is now needed.

Reply:
That's interesting. `Event` is just a better typed events map. Can we cast things `as EventEmitter`? Those classes _are_ compatible.

Reply:
You mean this?

```typescript
const emitter = new StrictEventEmitter()

// ...

await handleRequest(
  mockedRequest,
  handlers,
  {
    onUnhandledRequest: () => null,
  },
  emitter as EventEmitter, // <- here
  ...
)
```

this shows the error below.

```
Argument of type 'EventEmitter' is not assignable to parameter of type 'StrictEventEmitter<ServerLifecycleEventsMap>'.
  The types returned by 'eventNames()' are incompatible between these types.
    Type '(string | symbol)[]' is not assignable to type 'Event[]'.
      Type 'string | symbol' is not assignable to type 'Event'.
        'Event' could be instantiated with an arbitrary type which could be unrelated to 'string | symbol'.ts(2345)
```

Reply:
Strange but I don't see these errors while building/testing locally. It may be some TS versions discrepancy on your machine maybe? CI build/tests have also passed without type violations.